engine/src/main/java/org/hibernate/validator/engine/HibernateConstraintViolation.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.engine;
+
+import javax.validation.ConstraintViolation;
+
+/**
+ * A custom {@link ConstraintViolation} which allows to get additional information for a constraint violation.
+ *
+ * @since 5.3
+ */
+public interface HibernateConstraintViolation<T> extends ConstraintViolation<T> {
+	/**
+	 * @return an object that further describes the violation or {@code null}.
+	 */
+	Object getViolationContext();
```

Comment:
This is left to be done it seems?

+1 for the `public <C> C getViolationContext(Class<C> type)` signature, it leaves the door open for supporting several context objects, of different types (essentially the "type-safe heterogenous container" pattern from "Effective Java").

I don't like the method name though, how about `getPayload()`. To me, the method returns no context, just some data/payload retrieved from such context.